HBT.cpp

```diff
@@ -93,6 +94,8 @@ int main(int argc, char **argv)
 
     timer.Tick(world.Communicator);
     subsnap.AssignHosts(world, halosnap, partsnap);
+    MergerTreeInfo merger_tree;
+    merger_tree.StoreTracerIds(subsnap.Subhalos, HBTConfig.NumTracersForDescendants);
```

Comment:
I have found rare cases where SinkTrackId points at something unresolved. I think there's a tiny chance that two bound subhalos can be merged and then the combined set of particles is found to be unbound.  Maybe if we're unlucky with our initial choice of velocity frame for the unbinding?

If this happens then both progenitor objects would have DescendantTrackId=-1 because their most bound tracers are not in any bound halo. Unlike SinkTrackId, DescendantTrackId can only ever point at a resolved subhalo.